packages/firebase_vertexai/firebase_vertexai/lib/src/base_model.dart

```diff
@@ -87,11 +83,77 @@ abstract class BaseModel {
   static Uri _vertexUri(FirebaseApp app, String location) {
     var projectId = app.options.projectId;
     return Uri.https(
-      _baseUrl,
+      _baseAuthority,
       '/$_apiVersion/projects/$projectId/locations/$location/publishers/google',
     );
   }
 
+  final Uri _projectUri;
+  @override
+  final ({String prefix, String name}) model;
+
+  @override
+  String get baseAuthority => _baseAuthority;
+
+  @override
+  Uri taskUri(Task task) {
+    return _projectUri.replace(
+        pathSegments: _projectUri.pathSegments
+            .followedBy([model.prefix, '${model.name}:${task.name}']));
+  }
+}
+
+final class _GoogleAIUri implements _ModelUri {
+  _GoogleAIUri({
+    required String model,
+  })  : model = _normalizeModelName(model),
+        _baseUri = _googleAIBaseUri();
+
+  /// Returns the model code for a user friendly model name.
+  ///
+  /// If the model name is already a model code (contains a `/`), use the parts
+  /// directly. Otherwise, return a `models/` model code.
+  static ({String prefix, String name}) _normalizeModelName(String modelName) {
+    if (!modelName.contains('/')) return (prefix: 'models', name: modelName);
+    final parts = modelName.split('/');
+    return (prefix: parts.first, name: parts.skip(1).join('/'));
+  }
+
+  static const _apiVersion = 'v1beta';
+  // TODO - Update to vertex backend URI.
```

Comment:
@cynthiajoan - this is the most important thing I need help with - can you either try updating this to go through vertex, or provide instructions and vertex credentials for how to try it myself?

Reply:
Synced offline with test capability and url, I've made some modification and let me know what do you think about those.